include/boost/static_string/static_string.hpp

```diff
@@ -2608,6 +2608,119 @@ class basic_static_string
     const_iterator first,
     const_iterator last);
 
+  /** Erase from the string.
+
+      Erases all the characters from the string satisfying specific criteria.
+
+      @par Exception Safety
+
+      Strong guarantee.
+
+      @note All references, pointers, or iterators
+      referring to contained elements are invalidated. Any
+      past-the-end iterators are also invalidated.
+
+      @return `*this`
+
+      @param pred A unary predicate which returns ​true if the element should be erased.
+  */
+  template<typename UnaryPredicate>
+  BOOST_STATIC_STRING_CPP14_CONSTEXPR
+  basic_static_string&
+  erase_if(UnaryPredicate pred)
+  {
+    erase_if(data(), end(), pred);
+    return *this;
+  }
+
+  /** Erase from the string.
+
+      Erases `num` characters from the string, starting at `index`, satisfying specific criteria.
+      `num` is determined as the smaller of `count` and `size() - index`.
+
+      @par Exception Safety
+
+      Strong guarantee.
+
+      @tparam UnaryPredicate The type of the predicate.
+
+      @par Constraints
+
+      `UnaryPredicate` satisfies __UnaryPredicate__.
```

Comment:
is __UnaryPredicate__ defined in the qbk? It would go here:
https://github.com/boostorg/static_string/blob/develop/doc/qbk/main.qbk#L34